subsys/caf/modules/sensor_sampler.c

```diff
@@ -141,7 +321,9 @@ static int sensor_init(void)
 static void sample_thread_fn(void)
 {
 	int err = 0;
-	int64_t next_timeout;
+	int64_t sleep_time = INT64_MAX;
```

Comment:
start with sleep_time = 0 and no semaphore available.
as these are the true conditions - we don't want to sleep and we don't have any signal.